macos/Sources/Features/Terminal/TerminalController.swift

```diff
@@ -1023,6 +1028,32 @@ class TerminalController: BaseTerminalController, TabGroupCloseCoordinator.Contr
         }
     }
 
+    @IBAction func closeOtherTabs(_ sender: Any?) {
```

Comment:
I still want to review this function. This implementation is a bit odd to me and it's unclear this behaves well with the undo system. 

It seems AI written, honestly, but I'm sorry if its not 😝

Reply:
As a note: undo does work but it focuses the wrong tab on restore.

Reply:
Cleaned it up and made undo/redo work properly.